parcels/particleset.py

```diff
@@ -1247,6 +1249,22 @@ def _warn_outputdt_release_desync(outputdt: float, starttime: float, release_tim
             "Some of the particles have a start time difference that is not a multiple of outputdt. "
             "This could cause the first output of some of the particles that start later "
             "in the simulation to be at a different time than expected.",
-            FileWarning,
+            ParticleSetWarning,
             stacklevel=2,
         )
+
+
+def _warn_particle_times_outside_fieldset_time_bounds(time: np.ndarray, fieldset: FieldSet):
+    if np.any(time):
+        if np.any(time < fieldset.U.grid.time_full[0]) and not fieldset.U.allow_time_extrapolation:  # type: ignore[attr-defined]
+            warnings.warn(
+                "Some particles are set to be released before the fieldset's first time and allow_time_extrapolation is set to False.",
+                ParticleSetWarning,
+                stacklevel=2,
+            )
+        if np.any(time > fieldset.U.grid.time_full[-1]) and not fieldset.U.allow_time_extrapolation:  # type: ignore[attr-defined]
+            warnings.warn(
+                "Some particles are set to be released after the fieldset's last time and allow_time_extrapolation is set to False.",
+                ParticleSetWarning,
+                stacklevel=2,
+            )
```

Comment:
I think that this can be refactored to

```py
def _warn_particle_times_outside_fieldset_time_bounds(release_times: np.ndarray, time_full: np.ndarray):
    ...
```
and then the `if not allow_time_extrapolation` can be lifted up a level such that

```py
if not allow_time_extrapolation:
    _warn_particle_times_outside_fieldset_time_bounds(time, fieldset)
```

I think that would reduce the coupling of this function

Reply:
Good idea; just implemented!